tst/EnergyPlus/unit/EMSManager.unit.cc

```diff
@@ -826,6 +827,258 @@ TEST_F( EnergyPlusFixture, EMSManager_CheckIfAnyEMS_OutEMS ) {
+			"set Var8 = @RadToDeg 0.0,",
+			"set Var9 = @Exp 0.0,",
+			"set Var10 = @Ln 0.0,",
+			"set Var11 = @Max 0.0 1.0,",
+			"set Var12 = @Min 0.0 1.0,",
+			"set Var13 = @Abs 0.0,",
+			"set Var14 = @RANDOMUNIFORM 0.0 1.0,",
+			"set Var15 = @RANDOMNORMAL 0.0 1.0 2.0 3.0,",
+			"set Var16 = @SEEDRANDOM 0.0,",
+			"set Var17 = @RhoAirFnPbTdbW 0.0 1.0 2.0,",
+			"set Var18 = @CpAirFnWTdb 0.0 1.0,",
+			"set Var19 = @HfgAirFnWTdb 0.0 1.0,",
+			"set Var20 = @HgAirFnWTdb 0.0 1.0,",
+			"set Var21 = @TdpFnTdbTwbPb 0.0 1.0 2.0,",
+			"set Var22 = @TdpFnWPb 0.0 1.0,",
+			"set Var23 = @HFnTdbW 0.0 1.0,",
+			"set Var24 = @HFnTdbRhPb 0.0 1.0 2.0,",
+			"set Var25 = @TdbFnHW 0.0 1.0,",
+			"set Var26 = @RhovFnTdbRh 0.0 1.0,",
+			"set Var27 = @RhovFnTdbRhLBnd0C 0.0 1.0,",
+			"set Var28 = @RhovFnTdbWPb 0.0 1.0 2.0,",
+			"set Var29 = @RhFnTdbRhov 0.0 1.0,",
+			"set Var30 = @RhFnTdbRhovLBnd0C 0.0 1.0,",
+			"set Var31 = @RhFnTdbWPb 0.0 1.0 2.0,",
+			"set Var32 = @TwbFnTdbWPb 0.0 1.0 2.0,",
+			"set Var33 = @VFnTdbWPb 0.0 1.0 2.0,",
+			"set Var34 = @WFnTdpPb 0.0 1.0,",
+			"set Var35 = @WFnTdbH 0.0 1.0,",
+			"set Var36 = @WFnTdbTwbPb 0.0 1.0 2.0,",
+			"set Var37 = @WFnTdbRhPb 0.0 1.0 2.0,", // this looks suspicious
+			"set Var38 = @PsatFnTemp 0.0,",
+			"set Var39 = @TsatFnHPb 0.0 1.0,",
+			"set Var40 = @CpCW 0.0,",
+			"set Var41 = @CpHW 0.0,",
+			"set Var42 = @RhoH2O 0.0,",
+			"set Var43 = @SEVEREWARNEP 0.0,",
+			"set Var44 = @WARNEP 0.0,",
+			"set Var45 = @TRENDVALUE 0.0 1.0,",
+			"set Var46 = @TRENDAVERAGE 0.0 1.0,",
+			"set Var47 = @TRENDMAX 0.0 1.0,",
+			"set Var48 = @TRENDMIN 0.0 1.0,",
+			"set Var49 = @TRENDDIRECTION 0.0 1.0,",
+			"set Var50 = @TRENDSUM 0.0 1.0,",
+			"set Var51 = @CURVEVALUE 1 0.75;",
+
+//			"set Var52 = @TsatFnPb 0.0,", // not public in PsycRoutines so not available to EMS (commented out at line 2397 of RuntimeLanguageProcessor.cc)
+//			"set Var53 = @FATALHALTEP 0.0,", // terminates program, not unit test friendly
+
+		} );
+
+		ASSERT_FALSE( process_idf( idf_objects ) );
+
+		CheckIfAnyEMS(); // get EMS input
+		EMSManager::FinishProcessingUserInput = true;
+		bool ErrorsFound(false);
+		CurveManager::GetCurveInputData( ErrorsFound ); // process curve for use with EMS
+		EXPECT_FALSE( ErrorsFound );
+
+		bool anyRan;
+		EMSManager::ManageEMS( DataGlobals::emsCallFromHVACIterationLoop, anyRan );
+		EXPECT_TRUE( anyRan );
+
+		int index( 0 );
+		int offset( 24 );
+		EXPECT_EQ( DataRuntimeLanguage::ErlExpression( 1 ).Operator, FuncRound );
+		EXPECT_EQ( DataRuntimeLanguage::ErlExpression( 1 ).NumOperands, 1 );
+//		EXPECT_EQ( size( DataRuntimeLanguage::ErlExpression( 1 ).Operand ), 1 ); // why doesn't this compile? ... Array1D< ErlValueType > Operand
+		index = 1 + offset;
+		EXPECT_EQ( DataRuntimeLanguage::ErlVariable( index ).Name, "VAR1" );
+//		EXPECT_EQ( DataRuntimeLanguage::ErlVariable( 25 ).Value, 2.0 ); // why doesn't this compile? same issue as above ... ErlValueType Value
```

Comment:
I couldn't fully test out each EMS function since I couldn't figure out correct syntax for these